WaiterRobot/Ui/TableList/TableListScreen.swift

```diff
@@ -5,64 +5,78 @@ import UIPilot
 struct TableListScreen: View {
     @EnvironmentObject var navigator: UIPilot<Screen>
 
-    @StateObject private var strongVM = ObservableViewModel(vm: koin.tableListVM())
+    @StateObject private var viewModel = TableListObservableViewModel()
 
     private let layout = [
         GridItem(.adaptive(minimum: 100)),
     ]
 
-    // TODO: table has open/unpaid order indicator
     var body: some View {
-        unowned let vm = strongVM
-
-        let tableGroups = vm.state.tableGroups.data as? [TableGroup]
-        let tableGroupResource = onEnum(of: vm.state.tableGroups)
         VStack {
-            if let tableGroups, tableGroups.count > 1 {
-                TableListFilterRow(
-                    tableGroups: tableGroups,
-                    onToggleFilter: { vm.actual.toggleFilter(tableGroup: $0) },
-                    onSelectAll: { vm.actual.showAll() },
-                    onUnselectAll: { vm.actual.hideAll() }
-                )
-            }
+            switch onEnum(of: viewModel.state.tableGroupsArray) {
+            case let .error(resource):
+                Text(resource.exception.getLocalizedUserMessage())
+                    .foregroundStyle(.red)
 
-            RefreshableScrollView(for: tableGroupResource, onRefresh: { vm.actual.loadTables(forceUpdate: true) }) {
-                if case let .error(resource) = tableGroupResource {
-                    Text(resource.exception.getLocalizedUserMessage())
+            case let .loading(resource):
+                if resource.data == nil {
+                    ProgressView()
                 }
 
-                if tableGroups?.isEmpty == true {
-                    Text(localize.tableList.noTableFound())
-                        .multilineTextAlignment(.center)
-                        .frame(maxWidth: .infinity)
-                        .padding()
-                } else if let tableGroups {
-                    LazyVGrid(columns: layout) {
-                        ForEach(tableGroups.filter { !$0.hidden }, id: \.id) { group in
-                            if !group.tables.isEmpty {
-                                TableGroupSection(
-                                    tableGroup: group,
-                                    onTableClick: { vm.actual.onTableClick(table: $0) }
-                                )
-                            }
-                        }
-                    }
-                    .padding()
-                }
+            case .success:
+                EmptyView()
+            }
+
+            if let data = viewModel.state.tableGroupsArray.data {
+                let tableGroups = Array(data)
+                content(tableGroups: tableGroups)
```

Comment:
I would probably move the conversion to Array to the content function. So that in case you would need it at multiple places, you would not need to write the conversion each time.